misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -208,7 +211,12 @@ internal class SqsJobConsumer @Inject internal constructor(
                 )
                 Status.OK
               } catch (th: Throwable) {
-                log.error(th) { "error handling job from ${queue.queueName}" }
+                val mdcTags = TaggedLogger.getThreadLocalMdcContext()
```

Comment:
Seems like get and reset are always called in tandem; you could also have a method like `TaggedLogger.popThreadLocalMdcContext()` that does both (not picky about naming here).

Reply:
If left to my own un-imagination I would have probably named it `getAndResetThreadLocalMdcContext...` but pop is wildly better and on point thank you.